src/net/udp.rs

```diff
@@ -29,30 +31,71 @@ impl UdpSocket {
         let (len, _addr) = self.socket.recv_from(&mut self.recv_buffer)?;
 
         if len > 0 {
-            // TODO: Remove unwrap and funnel result error typesq
-            let packet: Packet = deserialize(&self.recv_buffer).unwrap();
+            // TODO: Remove unwrap and funnel result error types
+            let packet: Packet = deserialize(&self.recv_buffer[..len]).unwrap();
+            self.state.process_received(_addr, &packet);
             Ok(Some(packet))
         } else {
             Ok(None)
         }
     }
 
-    pub fn send(&mut self, packet: &Packet) -> io::Result<usize> {
-        let addr = packet.addr();
-        let buf = serialize(packet).unwrap();
+    pub fn send(&mut self, packet: Packet) -> io::Result<usize> {
+        let mut packet = packet;
```

Comment:
Do we need to rebind packet here, can we just pass it to preprocess?

Reply:
Was thinking that also. Yea we can make it `mut packet` as parameter.

Reply:
I think since we are moving it, it should be fine.